Gems/LmbrCentral/Code/Source/Shape/EditorBaseShapeComponent.cpp

```diff
@@ -129,35 +140,47 @@ namespace LmbrCentral
         m_shapeConfig = shapeConfig;
     }
 
-    AZ::Aabb EditorBaseShapeComponent::GetEditorSelectionBoundsViewport(
-        const AzFramework::ViewportInfo& /*viewportInfo*/)
+    AZ::Aabb EditorBaseShapeComponent::GetEditorSelectionBoundsViewport([[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo)
     {
         return GetWorldBounds();
     }
 
     bool EditorBaseShapeComponent::EditorSelectionIntersectRayViewport(
-        const AzFramework::ViewportInfo& /*viewportInfo*/,
-        const AZ::Vector3& src, const AZ::Vector3& dir, float& distance)
+        [[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo, const AZ::Vector3& src, const AZ::Vector3& dir, float& distance)
     {
         // if we are not drawing this or it is wireframe, do not allow selection
         if (!CanDraw() || !m_displayFilled)
         {
             return false;
         }
 
-        // Don't intersect with shapes when the camera is inside them
+        // don't intersect with shapes when the camera is inside them
         bool isInside = false;
         ShapeComponentRequestsBus::EventResult(isInside, GetEntityId(), &ShapeComponentRequests::IsPointInside, src);
         if (isInside)
         {
             return false;
         }
-        
+
         bool rayHit = false;
         ShapeComponentRequestsBus::EventResult(rayHit, GetEntityId(), &ShapeComponentRequests::IntersectRay, src, dir, distance);
         return rayHit;
     }
 
+    bool EditorBaseShapeComponent::SupportsEditorRayIntersect()
+    {
+        return AzToolsFramework::HelpersVisible();
+    }
+
+    bool EditorBaseShapeComponent::SupportsEditorRayIntersectViewport(const AzFramework::ViewportInfo& viewportInfo)
+    {
+        bool helpersVisible = false;
+        AzToolsFramework::ViewportInteraction::ViewportSettingsRequestBus::EventResult(
+            helpersVisible, viewportInfo.m_viewportId,
+            &AzToolsFramework::ViewportInteraction::ViewportSettingsRequestBus::Events::HelpersVisible);
+        return helpersVisible;
+    }
```

Comment:
Would there be a way to also support entity containers? Right now, we don't display helpers for entities that are descendants of closed entity containers. Sounds like this would only work for the "helpers disabled" case.

(It could be a follow-up change honestly, this works as a stand-alone change)